CMakeLists.txt

```
@@ -0,0 +1,270 @@
+include(GNUInstallDirs)
+include(CMakePackageConfigHelpers)
+include(sources.cmake)
+
+# The only direct cmake argument for now
+option(BUILD_SHARED_LIBS "Build shared library and only the shared library if \"ON\", default is static" OFF)
+
+#-----------------------------------------------------------------------------
+# Compose CFLAGS
+#-----------------------------------------------------------------------------
+
+# Some information copied from makefile_include.mk
+
+# Basic set
+set(LTM_C_FLAGS  -Wall -Wsign-compare -Wextra -Wshadow)
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wdeclaration-after-statement -Wbad-function-cast -Wcast-align)
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wstrict-prototypes -Wpointer-arith -Wsystem-headers)
+
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting build type to 'Release' as none was specified.")
+  set(CMAKE_BUILD_TYPE "Release")
+endif()
+
+set(CMAKE_C_FLAGS_DEBUG "-g3")
+set(CMAKE_C_FLAGS_RELEASE "-O3 -funroll-loops -fomit-frame-pointer")
+set(CMAKE_C_FLAGS_RELWITHDEBINFO "-g3 -O2")
+set(CMAKE_C_FLAGS_MINSIZEREL "-Os")
+
+if(COMPILE_LTO)
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS} -flto)
+    set(LTM_LD_FLAGS ${LTM_LD_FLAGS} -flto)
+endif()
+
+# What compiler do we have and what are their...uhm... peculiarities
+if(CMAKE_C_COMPILER_ID MATCHES "(C|c?)lang")
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS} -Wno-typedef-redefinition -Wno-tautological-compare -Wno-builtin-requires-header)
+    # Clang requires at least '-O1' for dead code eliminiation
+    set(CMAKE_C_FLAGS_DEBUG  "${CMAKE_C_FLAGS_DEBUG} -O1")
+endif()
+if(CMAKE_C_COMPILER MATCHES "mingw")
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS}  -Wno-shadow)
+endif()
+if(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS}  -Wno-nullability-completeness)
+endif()
+if(CMAKE_SYSTEM_NAME MATCHES "CYGWIN")
+    set(LTM_C_FLAGS  ${LTM_C_FLAGS} -no-undefined)
+endif()
+
+# TODO: coverage (lgcov)
+
+# If the user set the environment variables at generate-time, append them
+# in order to allow overriding our defaults.
+# ${LTM_CFLAGS} means the user passed it via sth like:
+# $ cmake -DLTM_CFLAGS="foo"
+set(LTM_C_FLAGS  ${LTM_C_FLAGS} ${LTM_CFLAGS})
+set(LTM_LD_FLAGS  ${LTM_LD_FLAGS} ${LTM_LDFLAGS})
+
+#-----------------------------------------------------------------------------
+# library target
+#-----------------------------------------------------------------------------
+add_library(${PROJECT_NAME}
+    ${SOURCES}
+)
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}>
+)
+
+target_compile_options(${PROJECT_NAME} BEFORE PRIVATE
+    ${LTM_C_FLAGS}
+)
+target_link_options(${PROJECT_NAME} BEFORE PRIVATE
+    ${LTM_LD_FLAGS}
+)
+
+set_target_properties(${PROJECT_NAME} PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
```

Set PUBLIC_HEADER property so the install with `PUBLIC_HEADER DESTINATION` picks it.
```suggestion
set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER
    ${PUBLIC_HEADERS}
)
```